code/scripting/api/libs/base.cpp

```diff
@@ -93,12 +93,17 @@ ADE_FUNC(rand32f,
 	float _max;
 	int numargs = ade_get_args(L, "|f", &_max);
 
-	float result = (float)Random::next() * Random::INV_F_MAX_VALUE;
+	// see also frand()
+	int num;
+	do {
+		num = Random::next();
+	} while (num == Random::MAX_VALUE);
```

Comment:
Are we sure there aren't values close to `MAX_AVLUE` like `MAX_VALUE - 1` that when multiplied by `INV_F_MAX_VALUE` will result in `1.0f` due to `float` imprecision?

If there are, we should exclude those values aw well.